sklearn/tree/_splitter.pyx

```diff
@@ -255,7 +274,18 @@ cdef class BaseDenseSplitter(Splitter):
 
         self.X = <DTYPE_t*> X_ndarray.data
         self.X_sample_stride = <SIZE_t> X.strides[0] / <SIZE_t> X.itemsize
-        self.X_fx_stride = <SIZE_t> X.strides[1] / <SIZE_t> X.itemsize
+        self.X_feature_stride = <SIZE_t> X.strides[1] / <SIZE_t> X.itemsize
+
+        if self.presort == 1:
+            self.X_idx_sorted = X_idx_sorted
+            self.X_idx_sorted_ptr = <INT32_t*> self.X_idx_sorted.data
+            self.X_idx_sorted_stride = (<SIZE_t> self.X_idx_sorted.strides[1] /
+                                        <SIZE_t> self.X_idx_sorted.itemsize)
```

Comment:
If we have a matrix with 10 samples and 5 features, X_feature_stride is 1 and X_sample_stride is 5. Feature stride is 1 because you only need to move over 1 to get to the next feature in the same sample. If you transpose this matrix, now X_sample_stride is 1, and X_feature_stride is 10. However, since you generate the index matrix from the appropriately oriented matrix, it's always has a feature stride of 10.